### PR TITLE
Replace endpoint terminology

### PR DESCRIPTION
**Pull request by acoburn**
Addresses #30 

This swaps out the use of `endpoint` for more appropriate terms: resource, URI, URL, API and the like.

These changes were applied to the Protocol document along with the following individual subscription types:

* WebSockets
* EventSource
* StreamingHTTPFetch
* WebHooks

I left most of the usages of "endpoint" in WebHooks in place, as much of that data model was built on those terms, and I'd rather let @jaxoncreed comment on that.

### CODE REVIEW

File: protocol.html [hunk 1]
```diff
@@ -534,16 +534,16 @@ <h3 property="schema:name">Notification Subscription</h3>
...
 
                   <p>All other fields are defined by the particular subscription type. Some common features are described in the Features section of this document.</p>
 
                   <p>An example <code>POST</code> request using a <code>DPoP</code> bound access token is below:</p>
 
                   <figure id="notification-subscription-request" class="example listing" rel="schema:hasPart" resource="#notification-subscription-request">
-                    <p class="example-h"><span>Example</span>: Requesting a WebSocket subscription endpoint from the Notification Subscription.</p>
+                    <p class="example-h"><span>Example</span>: Requesting a WebSocket subscription URI from the Notification Subscription API.</p>
```
Comment (elf-pavlik) [inline protocol.html:546]:
```suggestion
                    <p class="example-h"><span>Example</span>: Requesting a WebSocket subscription from the  Subscription API.</p>
```

File: protocol.html [hunk 2]
```diff
@@ -562,29 +562,29 @@ <h3 property="schema:name">Notification Subscription</h3>
                     <figcaption property="schema:name"><code>POST</code> request including <code>type</code> and <code>topic</code> targeting the Notification Subscription.</figcaption>
                   </figure>
 
-                  <p>A successful response will contain a URL to the subscription endpoint that can be used directly with a JavaScript client.</p>
+                  <p>A successful response will contain a subscription URI that can be used directly with a JavaScript client.</p>
 
                   <figure id="notification-subscription-response" class="example listing" rel="schema:hasPart" resource="#notification-subscription-response">
-                    <p class="example-h"><span>Example</span>: Response from the Notification Subscription including the WebSocket subscription endpoint.</p>
+                    <p class="example-h"><span>Example</span>: Response from the Notification Subscription including the WebSocket subscription URI.</p>
```
Comment (elf-pavlik) [inline protocol.html:568]:
```suggestion
                    <p class="example-h"><span>Example</span>: Response from the Notification Subscription including the WebSocket notifications source.</p>
```

File: protocol.html [hunk 3]
```diff
@@ -562,29 +562,29 @@ <h3 property="schema:name">Notification Subscription</h3>
...
-                    <figcaption property="schema:name">Response to the <code>POST</code> request including <code>type</code> and <code>endpoint</code>.</figcaption>
+                    <figcaption property="schema:name">Response to the <code>POST</code> request including <code>type</code> and <code>subscription</code>.</figcaption>
                   </figure>
 
-                  <p>In JavaScript, a client can use the data in the response to establish a connection to the WebSocket endpoint.</p>
+                  <p>In JavaScript, a client can use the data in the response to establish a connection to the WebSocket server.</p>
 
                   <figure id="websocket-new" class="example listing" rel="schema:hasPart" resource="#websocket-new">
-                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection to a subscription endpoint.</p>
+                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection with a subscription URI.</p>
```
Comment (elf-pavlik) [inline protocol.html:585]:
```suggestion
                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection with a notifications source.</p>
```

File: protocol.html [hunk 4]
```diff
@@ -562,29 +562,29 @@ <h3 property="schema:name">Notification Subscription</h3>
...
 <code></code>
 <code>{</code>
 <code>  "@context": "https://www.w3.org/ns/solid/notification/v1",</code>
 <code>  "type": "WebSocketSubscription2021",</code>
-<code>  "endpoint": "wss://websocket.example/?auth=Ys3KiUq"</code>
+<code>  "source": "wss://websocket.example/?auth=Ys3KiUq"</code>
 <code>}</code></pre>
 
-                    <figcaption property="schema:name">Response to the <code>POST</code> request including <code>type</code> and <code>endpoint</code>.</figcaption>
+                    <figcaption property="schema:name">Response to the <code>POST</code> request including <code>type</code> and <code>subscription</code>.</figcaption>
```
Comment (elf-pavlik) [inline protocol.html:579]:
```suggestion
                    <figcaption property="schema:name">Response to the <code>POST</code> request including subscription <code>type</code> and notifications <code>source</code>.</figcaption>
```

File: protocol.html [hunk 5]
```diff
@@ -562,29 +562,29 @@ <h3 property="schema:name">Notification Subscription</h3>
...
 
-                  <p>In JavaScript, a client can use the data in the response to establish a connection to the WebSocket endpoint.</p>
+                  <p>In JavaScript, a client can use the data in the response to establish a connection to the WebSocket server.</p>
 
                   <figure id="websocket-new" class="example listing" rel="schema:hasPart" resource="#websocket-new">
-                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection to a subscription endpoint.</p>
+                    <p class="example-h"><span>Example</span>: Establishing a new WebSocket connection with a subscription URI.</p>
 
-                    <pre about="#websocket-new" property="schema:description" typeof="fabio:Script"><code>const ws = new WebSocket(endpoint, type)</code></pre>
+                    <pre about="#websocket-new" property="schema:description" typeof="fabio:Script"><code>const ws = new WebSocket(subscription, type)</code></pre>
```
Comment (elf-pavlik) [inline protocol.html:587]:
```suggestion
                    <pre about="#websocket-new" property="schema:description" typeof="fabio:Script"><code>const ws = new WebSocket(source, type)</code></pre>
```

**Review by elf-pavlik — COMMENTED** [review]
I think this is great, still, IMO we should improve it.

I made some inline suggestions to use two terms in a consistent way
1. Subscription API - HTTP URL where the subscription is created via an authenticated POST
2. Notifications Source - URL where the actual notifications are streamed from

Since it is a notifications protocol spec I think we don't need to use *Notifications Subscription* since there will be no other subscriptions here. On the other hand, we should keep a clear distinction between the subscription and the notifications, as discussed during last week's meeting.

I didn't suggest any changes on specific notification types since I believe they should not be included in this document at all.

File: streaming-http-subscription-2021.bs [hunk 6]
```diff
@@ -71,8 +71,8 @@ An StreamingHTTPSubscription2021 API SHOULD support the [Solid Notifications Fea
 
 The StreamingHTTPSubscription2021 type defines the following properties:
 
-: endpoint
-:: The endpoint property is used in the body of the subscription response.
+: subscription
+:: The subscription property is used in the body of the subscription response.
```
Comment (elf-pavlik) [inline streaming-http-subscription-2021.bs:75]:
```suggestion
: source
:: The source property is used in the body of the subscription response.
```

File: protocol.html [hunk 7]
```diff
@@ -655,11 +655,11 @@ <h3 property="schema:name">EventSourceSubscription2021</h3>
...
+                    <dt about="#eventsourcesubscription2021-subscription" property="skos:prefLabel" typeof="skos:Concept"><dfn id="eventsourcesubscription2021-subscription">subscription</dfn></dt>
+                    <dd about="#eventsourcesubscription2021-subscription" property="skos:definition"><span about="" id="eventsourcesubscription2021-subscription-https" rel="spec:requirement" resource="#eventsourcesubscription2021-subscription-https"><span property="spec:statement">The <code>subscription</code> property is used in the body of the subscription response. The value of this property <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be a URI, using the <code>https</code> scheme.</span></span> A JavaScript client would use the entire value as input to the EventSource constructor.</dd>
```
Comment (elf-pavlik) [inline protocol.html:662]:
```suggestion
                    <dt about="#eventsourcesubscription2021-source" property="skos:prefLabel" typeof="skos:Concept"><dfn id="eventsourcesubscription2021-source">source</dfn></dt>
                    <dd about="#eventsourcesubscription2021-source" property="skos:definition"><span about="" id="eventsourcesubscription2021-subscription-https" rel="spec:requirement" resource="#eventsourcesubscription2021-source-https"><span property="spec:statement">The <code>source</code> property is used in the body of the subscription response. The value of this property <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be a URI, using the <code>https</code> scheme.</span></span> A JavaScript client would use the entire value as input to the EventSource constructor.</dd>
```

File: protocol.html [hunk 8]
```diff
@@ -672,8 +672,8 @@ <h3 property="schema:name">LinkedDataNotificationsSubscription2021</h3>
...
+                    <dd about="#linkeddatanotificationssubscription2021-subscription" property="skos:definition"><span about="" id="linkeddatanotificationssubscription2021-subscription-https" rel="spec:requirement" resource="#linkeddatanotificationssubscription2021-subscription-https"><span property="spec:statement">The <code>subscription</code> property is used in the body of the subscription request. The value of this property <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be a URI, using the <code>https</code> scheme.</span></span> <span about="" id="linkeddatanotificationssubscription2021-subscription-ldn-receiver" rel="spec:requirement" resource="#linkeddatanotificationssubscription2021-subscription-ldn-receiver"><span property="spec:statement">The inbox URI <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> conform to the <a href="https://www.w3.org/TR/ldn/#receiver" rel="cito:citesAsAuthority">receiver</a> portion of the Linked Data Notifications specification.</span></span></dd>
```
Comment (elf-pavlik) [inline protocol.html:676]:
```suggestion
                    <dt about="#linkeddatanotificationssubscription2021-inbox" property="skos:prefLabel" typeof="skos:Concept"><dfn id="linkeddatanotificationssubscription2021-inbox">inbox</dfn></dt>
                    <dd about="#linkeddatanotificationssubscription2021-inbox" property="skos:definition"><span about="" id="linkeddatanotificationssubscription2021-inbox-https" rel="spec:requirement" resource="#linkeddatanotificationssubscription2021-inbox-https"><span property="spec:statement">The <code>inbox</code> property is used in the body of the subscription request. The value of this property <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be a URI, using the <code>https</code> scheme.</span></span> <span about="" id="linkeddatanotificationssubscription2021-subscription-ldn-receiver" rel="spec:requirement" resource="#linkeddatanotificationssubscription2021-subscription-ldn-receiver"><span property="spec:statement">The inbox URI <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> conform to the <a href="https://www.w3.org/TR/ldn/#receiver" rel="cito:citesAsAuthority">receiver</a> portion of the Linked Data Notifications specification.</span></span></dd>
```

**Review by elf-pavlik — APPROVED** [review]
LGTM

I could follow up with PR removing details or each subscription type from the protocol document.

**Comment by jaxoncreed** [comment]
I don't have a strong preference for terminology. If redefining "endpoint" to something like "Target URL" makes sense in the context of the rest of the spec, I don't think it would be too confusing for the webhook spec.

**Comment by acoburn** [comment]
We may iterate a bit further on this topic, but this gets us pretty far along for now

**Pull request merged**